marshmallow/schema.py

```diff
@@ -99,6 +101,37 @@ def __new__(mcs, name, bases, attrs):
         for attr in mcs.FUNC_LISTS:
             attr_copy = copy.copy(getattr(klass, attr))
             setattr(klass, attr, attr_copy)
+
+        # Now add in the decorated processors. By doing this after constructing
+        # the class, we let standard inheritance do all the hard work.
+        mro = klass.mro()
+
+        klass.processors = defaultdict(list)
+        for attr_name in dir(klass):
+            # Need to look up the actual descriptor, not whatever might be
+            # bound to the class.
+            for parent in mro:
+                try:
+                    attr = parent.__dict__[attr_name]
+                except KeyError:
+                    continue
+                else:
+                    break
+            else:
+                # In case we didn't find the attribute and didn't break above.
+                # This isn't necessary and we should never hit this.
```

Comment:
I need to use `clazz.__dict__` to access the descriptor. I need to explicitly traverse the MRO to handle inheritance correctly, because inherited descriptors are not on `clazz.__dict__` and need to be pulled from the parent.

The `else` clause on the `for` loop is just for completeness. Even though that branch should be impossible to hit, I still didn't want something where `attr` could potentially end up undefined.